### ISSUE TITLE
Github login

### ISSUE THREAD

**Issue by roydekleijn** [heresy/angular2-social-login#20]
Hi,

It would be create if we can add Github login.

Can you give me some pointers on how to extend this awesome feature.

Roy

**Comment by roydekleijn** [comment]
can we pls collaborate on this ?

**Comment by kpsrinivas** [comment]
Sure. Let me know if you have a push ready.

**Comment by ip4368** [comment]
I think I know how the library works. I think I can implement it. Just wondering would you like Github App or OAuth for Github integration? For just basic info like email and name, OAuth should work, just depend on which one you are looking for.

**Comment by roydekleijn** [comment]
OAuth would be super awesome!

**Comment by ip4368** [comment]
I just realized that the OAuth 2 that Github provided didn't support implicit flow. Creating a fully client side auth will sacrifice security. This may be solved by implementing a server side end point, but will not be fully client side. So I guess this is not very possible in this project, at least for now, until Github support implicit flow.

**Comment by roydekleijn** [comment]
Does the same apply for  Github App ?

**Comment by ip4368** [comment]
Yeah, apparently apparently their Github app integration server is the resource server, and OAuth is the authentication server (which make sense in OAuth architecture), so all authentication basically need to go through the OAuth server.
